common.sh

```diff
@@ -187,9 +187,39 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    local bash_option=$-
+    local restore_e=0
+    local get_instance_ip_succeed=0
+    local instance_ips_array=()
+    if [[ $bash_option =~ e ]]; then
+        restore_e=1
+        set +e
+    fi
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        instance_ips_array=($INSTANCE_IPS)
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ] && [ "${#instance_ips_array[@]}" == "${#INSTANCE_IDS[@]}" ]; then
```

Comment:
I made a more detailed comment below, but you should be consistent in your use of quotes with eq.  Given the heavy use of bash-ism in the last conditional, I also have to ask why you're using sh-style checks instead of bash-style checks.

Reply:
> Given the heavy use of bash-ism in the last conditional, I also have to ask why you're using sh-style checks instead of bash-style checks.

Could you elaborate how the last conditional can be modified to be a bash-style check?

Reply:
```
if [[ $ret == 0 && -n $instance_ips && ${#instance_ips_array[@]} -eq $${#INSTANCE_IDS[@]} ]]; then
```

Note that I did not directly test that, so make sure it's right.